pyhf/utils.py

```diff
@@ -0,0 +1,27 @@
+import json, jsonschema
+import pkg_resources
+
+from .exceptions import InvalidSpecification
+
+DEFAULT_SCHEMA = pkg_resources.resource_filename(__name__,'data/spec.json')
+def get_default_schema():
+  global DEFAULT_SCHEMA
+  return DEFAULT_SCHEMA
+
+SCHEMA_CACHE = {}
+def load_schema(schema):
+    global SCHEMA_CACHE
+    try:
+        return SCHEMA_CACHE[schema]
+    except KeyError:
+        pass
+
```

Comment:
do we really want to swallow this error?

Reply:
This error occurs when the provided schema wasn't cached. In this case, it's because it wasn't loaded in the cache yet -- so load the json, cache it, and return it. This is just a try/except to be more pythonic.

Reply:
This is to handle the case where the schema is very large and there's also no reason why we shouldn't cache the schemas used for a given instance of `pyhf`.

Reply:
ah sorry I was confused by the naming. I would rename this to `load_schema_from_cache` or `schema_from_cache`

Reply:
scatch that. I just looked at the code. it's fine.